Modules/_queuemodule.c

```diff
@@ -263,8 +258,29 @@ simplequeue_new_impl(PyTypeObject *type)
 
     return (PyObject *) self;
 }
+typedef struct {
+    int handed_off;
+    simplequeueobject *queue;
+    PyObject *item;
+} HandoffData;
+
+static void
+maybe_unparked_thread(HandoffData *data, PyObject **item, int has_more_waiters)
+{
+    if (item == NULL) {
+        // Didn't unpark a thread
+        data->handed_off = 0;
+    }
+    else {
+        // Successfully unparked a thread
+        *item = data->item;
+        data->handed_off = 1;
+    }
+    data->queue->has_threads_waiting = has_more_waiters;
+}
```

Comment:
I think what bothers me about the naming here, is that it is focused on the possible waiting thread. Perhaps it would be clearer to name this based on the actual queue operation?

Also, perhaps also consider using a `bool` datatype for `HandoffData.handed_off` (nit).

Feel free to ignore this, if you think I'm totally wrong :)

```suggestion
check_data_handoff(HandoffData *data, PyObject **item, bool has_more_waiters)
{
    // Check if data was handed off to a waiting thread.
    if (item == NULL) {
        data->handed_off = false;
    }
    else {
        *item = data->item;
        data->handed_off = true;
    }
    data->queue->has_threads_waiting = has_more_waiters;
}
```

Reply:
> Perhaps it would be clearer to name this based on the actual queue operation?

This looks good to me! 

> Also, perhaps also consider using a bool datatype for HandoffData.handed_off (nit).

Yeah, this is better. I'm still not fully used to using `bool` in C :)

Reply:
The `has_more_waiters` function parameter should be an `int`. It needs to match the signature of `_Py_unpark_fn_t`. (We could change that signature to use a `bool`, but maybe in a separate PR?)

Reply:
> I'm still not fully used to using bool in C :)

Me neither 😄

> The `has_more_waiters` function parameter should be an `int`. It needs to match the signature of `_Py_unpark_fn_t`. (We could change that signature to use a `bool`, but maybe in a separate PR?)

For file scope, `stdbool.h` stuff is ok, but we don't want it to accidentally leak out through Python.h again[^1]. Let's be conservative and keep the parking lot API on `int`.


[^1]: #90904